implementations/rust/ockam/ockam_api/tests/authority.rs

```diff
@@ -0,0 +1,390 @@
+
+    let workers = ctx.list_workers().await?;
+
+    assert!(!workers.contains(&Address::from(DefaultAddress::DIRECT_AUTHENTICATOR)));
+
+    ctx.stop().await?;
+
+    Ok(())
+}
+
+struct Admin {
+    identifier: Identifier,
+    client: DirectAuthenticatorClient,
+}
+
+async fn setup(
+    ctx: &Context,
+    secure_channels: Arc<SecureChannels>,
+    number_of_admins: usize,
+) -> Result<Vec<Admin>> {
+    use ockam_core::compat::collections::HashMap;
+    let now = now()?;
+
+    let mut admin_ids = vec![];
+
+    let mut trusted_identities = HashMap::<Identifier, AttributesEntry>::new();
+
+    let mut attrs = BTreeMap::<Vec<u8>, Vec<u8>>::new();
+    attrs.insert(b"ockam-role".to_vec(), b"enroller".to_vec());
+    attrs.insert(b"trust_context_id".to_vec(), b"123456".to_vec());
+
+    for _ in 0..number_of_admins {
+        let admin = secure_channels
+            .identities()
+            .identities_creation()
+            .create_identity()
+            .await?
+            .identifier()
+            .clone();
+
+        let entry = AttributesEntry::new(attrs.clone(), now, None, None);
+        trusted_identities.insert(admin.clone(), entry);
+
+        admin_ids.push(admin);
+    }
+
+    let mut configuration = default_configuration().await?;
+
+    configuration.no_direct_authentication = false;
+
+    configuration.trusted_identities = PreTrustedIdentities::Fixed(trusted_identities);
+
+    authority_node::start_node(ctx, &configuration).await?;
+
+    let mut admins = vec![];
+    for admin_id in admin_ids {
+        let sc = secure_channels
+            .create_secure_channel(ctx, &admin_id, route!["api"], SecureChannelOptions::new())
+            .await?;
+
+        let client = DirectAuthenticatorClient::new(
+            RpcClient::new(route![sc, DefaultAddress::DIRECT_AUTHENTICATOR], ctx).await?,
+        );
+
+        admins.push(Admin {
+            identifier: admin_id,
+            client,
+        });
+    }
+
+    Ok(admins)
+}
+
+#[ockam_macros::test]
+async fn test_only_admin_has_access(ctx: &mut Context) -> Result<()> {
+    use std::collections::HashMap;
+
+    let secure_channels = secure_channels();
+
+    let admins = setup(ctx, secure_channels.clone(), 1).await?;
+    let admin = &admins[0];
+
+    let member = secure_channels
+        .identities()
+        .identities_creation()
+        .create_identity()
+        .await?
+        .identifier()
+        .clone();
+
+    let mut attributes = HashMap::<&str, &str>::default();
+    attributes.insert("key", "value");
+
+    admin.client.add_member(member.clone(), attributes).await?;
+
+    let sc = secure_channels
+        .create_secure_channel(ctx, &member, route!["api"], SecureChannelOptions::new())
+        .await?;
+
+    let member_client = DirectAuthenticatorClient::new(
+        RpcClient::new(route![sc, DefaultAddress::DIRECT_AUTHENTICATOR], ctx).await?,
+    );
+
+    let timeout = Arc::new(AtomicBool::new(true));
```

Comment:
The name "timeout" is weird here. It's more like `list_is_accessible_for_non_admin` which should not switch from `false` to `true`

Reply:
The problem is that we don't have a proper error handling for such case (request is just dropped at Incoming ABAC), therefore this call if unsuccessful - just times out. Hence the name.

Reply:
I understand the implementation :-). I think it would read better with a different name. Not a big deal if you want to leave it like that.